src/lib/nvme_pt_ioctl.c

```diff
@@ -727,6 +733,128 @@ static int validate_sessn(struct opal_device *dev)
 	return 0;
 }
 
+static int parse_tper_host_prop(struct sed_device *device)
+{
+	struct opal_device *dev = device->priv;
+	int payload_len = dev->payload.len;
+	struct sed_tper_properties *tper = &device->discv.sed_tper_props;
+
+	/* TPer properties are returned as key-value pairs */
+	for (int i = 0; i < payload_len; i++) {
+		if(resp_token_match(dev->payload.tokens[i], OPAL_STARTNAME)) {
+			int jmp = get_payload_string(dev, i + 1);
+			int key_len = dev->payload.tokens[i + 1]->len - jmp;
+			char *key = malloc(key_len);
+			if (!key)
+				return -ENOMEM;
+
+			memset(key, 0, key_len);
+			memcpy(key, dev->payload.tokens[i + 1]->pos + jmp, key_len);
+
+			if (strncmp("MaxComPacketSize", key, key_len) == 0)
```

Comment:
this could also be simplfied. One idea if sed_tper_properties is defined as follows:
struct sed_tper_properties {
     struct  {
         char *name;
         uint64_t value;
     } property[23];
};

then this routine could populate property array instead of having massive if else block

Reply:
I'm not just trying to populate here...I'm trying to do a key match here. hence comparing them.

Reply:
The number of TPer properties returned may vary as spec says. But for time being we can assign a higher value and use it.